### PR TITLE
 Added post/comment reporting UI

### LINKED ISSUE #235: UI for Reporting Posts & Comments
This issue is just for the UI for reporting a post or a comment. It does not include a UI for moderators to be notified of or to act on reported comments. 

Design is below. 

API issue is #197

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### What are the relevant tickets?
Fixes #235 

#### What's this PR do?
Adds link and confirmation dialog to report posts and comments

#### How should this be manually tested?
Go to a post, verify that you can report both posts and comments and that the validation on the input displays and doesn't let you submit until you've fixed it.

### CODE REVIEW

**Comment by aliceriot** [comment]
functionality looks good! reading code now

File: static/js/lib/validation.js [hunk 1]
```diff
@@ -75,6 +75,19 @@ export const validateChannelEditForm = validate([
   )
 ])
 
+export const validateContentReportForm = validate([
+  validation(
+    R.compose(R.gt(R.__, 100), R.length),
```
Comment (aliceriot) [inline static/js/lib/validation.js]: I think this could just be `R.lt(100)`

File: static/js/lib/validation.js [hunk 2]
```diff
@@ -75,6 +75,19 @@ export const validateChannelEditForm = validate([
...
+    R.lensPath(["value", "reason"]),
+    "Reason length is limited to 100 characters"
+  ),
+  validation(
+    R.compose(R.lt(R.__, 3), R.length),
```
Comment (aliceriot) [inline static/js/lib/validation.js]: similarly this could be `R.gt(3)`

**Comment by rhysyngsun** [comment]
@aliceriot feedback addressed

**Pull request merged**